crates/core/src/surfnet/svm.rs

```diff
@@ -937,91 +944,245 @@ impl SurfnetSvm {
+                            VersionedMessage::Legacy(legacy_msg) => (
+                                legacy_msg.header,
+                                legacy_msg
+                                    .account_keys
+                                    .iter()
+                                    .map(|k| k.to_string())
+                                    .collect(),
+                                legacy_msg
+                                    .instructions
+                                    .iter()
+                                    .map(|ix| UiCompiledInstruction::from(ix, None))
+                                    .collect(),
+                                None,
+                            ),
+                            VersionedMessage::V0(v0_msg) => (
+                                v0_msg.header,
+                                v0_msg.account_keys.iter().map(|k| k.to_string()).collect(),
+                                v0_msg
+                                    .instructions
+                                    .iter()
+                                    .map(|ix| UiCompiledInstruction::from(ix, None))
+                                    .collect(),
+                                Some(
+                                    v0_msg
+                                        .address_table_lookups
+                                        .iter()
+                                        .map(UiAddressTableLookup::from)
+                                        .collect(),
+                                ),
+                            ),
+                        };
+
+                    let inner_instructions = OptionSerializer::Some(
+                        meta.inner_instructions
+                            .iter()
+                            .enumerate()
+                            .map(|(i, inner_ixs)| UiInnerInstructions {
+                                index: i as u8,
+                                instructions: inner_ixs
+                                    .iter()
+                                    .map(|ix| {
+                                        UiInstruction::Compiled(UiCompiledInstruction::from(
+                                            &ix.instruction,
+                                            Some(ix.stack_height as u32),
+                                        ))
+                                    })
+                                    .collect(),
+                            })
+                            .collect(),
+                    );
+
+                    let return_data = match meta.return_data.data.is_empty() {
+                        true => OptionSerializer::None,
+                        false => OptionSerializer::Some(UiTransactionReturnData {
+                            program_id: meta.return_data.program_id.to_string(),
+                            data: (
+                                base64::encode(&meta.return_data.data),
```

Comment:
Looks like this `encode` function is deprecated, can you use the recommended `Engine::decode` instead?